### ISSUE TITLE
[Licensing] Legality of including SteamOS Compositor in other distributions

### ISSUE THREAD

**Issue by btegs** [ValveSoftware/SteamOS#610]
The SteamOS Compositor and its dependencies are currently a great way to boot directly into Steam. Since [there is a PPA that brings it to Ubuntu](https://launchpad.net/~mdeslaur/+archive/ubuntu/steamos), are there any licensing or any other legal issues for the package maintainers of other distros like Arch, Fedora and Solus to include it? These distributions include the Steam runtime in their non-free repositories and it would be nice if they could include the `steamos-compositor` and `steamos-modeswitch-inhibitor` packages.

If it is ok, can these be included as-is with the Steam branding or do they have to strip it down into a generic, non-branded version?

**Comment by Plagman** [comment]
It's all MIT-licensed code, you can use it in any way you like as long as the license is carried over.

**Comment by btegs** [comment]
So if there is no issue with the license or branding, then I'd love to see it in something like Solus. [I did ask the head of the project](https://www.reddit.com/r/SolusProject/comments/6qs8zw) about it, but he was unsure. But if you said we are good to go, then thanks! @ikeydoherty @solus-project

**Comment by ikeydoherty** [comment]
Random ping. So the code isn't the part I was concerned about (I know most of this is forks of xcompmgr, etc.) - I'm more concerned about the Steam + Valve **branding** being used, not the code.

**Comment by johnv-valve** [comment]
I wouldn't recommend taking legal advice from github issue discussions, but branding generally falls under trademark law which is distinct from copyright law.

**Comment by btegs** [comment]
I see that @johnv-valve just closed this. So can other distros use the compositor as-is or does it have to be stripped of all branding. That is what Ikey is curious about.

**Comment by btegs** [comment]
One other thing... if you say this isn't the right place and choose not to answer, who do I contact regarding this?

**Comment by btegs** [comment]
https://aur.archlinux.org/packages/steamos-compositor/

Arch is including it and the legal team hasn't gone after them.

**Comment by ikeydoherty** [comment]
Arch's repo isn't exactly a solid indication of that, and do note that's the AUR - the `U` meaning User. It's not part of the core Arch Linux repositories.

Due to the murkiness that still shrouds this issue (i.e. closed issue which raised more questions than it answered) I'm personally not going to permit the Steam OS components into Solus unless they're completely devoid of branding. They'll all need patching to actually be allowed in the repos in the first place, and common sense says if we patch it, we can no longer say its the original thing that SteamOS has, thus can't be branded as such. (Even though thats what the composite manager is already, patched/forked of another. :P)

**Comment by johnv-valve** [comment]
Ok I don't know what 'murkiness' you are talking about. @Plagman answered very clearly "It's all MIT-licensed code, you can use it in any way you like as long as the license is carried over." So it is completely legal to include this code in your distribution and that is why I closed this issue. Also completely legal to modify this code in any way you like.

Now if you wanted to rename your distribution to "SteamOS" or "SteamSOLUS" or "ValveOS" then that would be a branding issue. Any copyright license is irrelevant as that becomes a trademark issue since you would be misleading people. 

I am no lawyer, but I don't think shipping a package named "steamos-compositor" violates a trademark, any more than shipping a package named "linux-latest" violates Linus' trademark.

**Comment by btegs** [comment]
I've come to realize that in order for more distros to use it, it has to be forked, stripped of branding, given a name like `gameos-compositor`and put up on its own Git repo.

The compositor works really well in Debian/Ubuntu and if it can be independently developed and brought to other distros (or non Steam services) in this matter, then the better.

Thanks for the info Ikey. :-)

**Comment by ikeydoherty** [comment]
@btegs that isn't what @johnv-valve said and none of this pertains to why I rejected entry for them in Solus.

@johnv-valve  I was actually talking about visual branding and trademarks, i.e. for the SteamOS session/compositor/bits. Thought that much was obvious, my apologies. As for renaming the distribution, I like Steam, but not **that** much :P I'm fully aware of the licensing on the **code** itself, as I already said in my first comment, and that my query had nothing to do with code.

Anywho - it's really not a big deal, I got randomly pinged, I answered. I don't currently see _much_ in the way of demand for these components in Solus (though I'm happy to reevaluate in time if necessary) - and right now our package inclusion policy wouldn't permit entry to some of those components.

Let's table this for now.

**Comment by btegs** [comment]
@ikeydoherty - Thanks for taking the time in responding to this bug/request. The compositor seemed like a good idea since I got better results on Ubuntu with it and figured that it would transition to Solus as well. It can be nice logging directly into a game focused compositor instead of having the OS overhead and letting the session's compositor try its best.

In a nutshell, I got less tearing and shuddering with Ubuntu 16.04 + SteamOS Compositor compared with the Budgie version of Solus.
